### PR TITLE
Workflow updates

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

**Comment by yutannihilation** [comment]
Does this supersede #4697?

**Comment by yutannihilation** [comment]
It seems so, thanks!

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -26,78 +22,49 @@ jobs:
...
       - uses: actions/checkout@v2
 
-      - uses: r-lib/actions/setup-r@master
+      - uses: r-lib/actions/setup-pandoc@v2
+
+      - uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.config.r }}
           http-user-agent: ${{ matrix.config.http-user-agent }}
+          use-public-rspm: true
```
Comment (yutannihilation) [inline .github/workflows/R-CMD-check.yaml:58]: I think we should not use RSPM for R-devel. Using the release binary tends to cause the "Graphics API version mismatch" error, if I remember correctly.

**Review by yutannihilation — COMMENTED** [review]
Looks good in general. Note that, checks on Ubuntu runner will probably keep failing because it fails to install sf (c.f. https://github.com/r-spatial/sf/issues/1899). I thought this would be resolved soon, but we might need to consider installing the dev version for now.

File: _pkgdown.yml [hunk 2]
```diff
@@ -2,21 +2,16 @@ url: https://ggplot2.tidyverse.org
...
 development:
   mode: auto
 
 authors:
-  Winston Chang:
-    href: https://github.com/wch
-  Lionel Henry:
-    href: https://github.com/lionel-
-  Thomas Lin Pedersen:
-    href: https://data-imaginist.com
```
[the remark this answers is not in the document]
Comment (thomasp85) [reply _pkgdown.yml:19]: We have these links internally now

**Comment by thomasp85** [comment]
@yutannihilation has had more experience with the actions for ggplot2 so I'll refer to him and his comments for review

File: .github/workflows/R-CMD-check.yaml [hunk 3]
```diff
@@ -26,78 +22,49 @@ jobs:
...
-      - name: Install dependencies
-        run: |
-          remotes::install_deps(dependencies = TRUE)
-          remotes::install_cran("rcmdcheck")
-        shell: Rscript {0}
-
-      - name: Check
-        run: rcmdcheck::rcmdcheck(args = "--no-manual", error_on = "warning", check_dir = "check")
-        shell: Rscript {0}
+          extra-packages: any::rcmdcheck
```
[the remark this answers is not in the document]
Comment (hadley) [reply .github/workflows/R-CMD-check.yaml:65]: FWIW these should go in DESCRIPTION in `Config/needs/check`

File: .github/workflows/R-CMD-check.yaml [hunk 4]
```diff
@@ -26,78 +22,49 @@ jobs:
...
+
+      - uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.config.r }}
           http-user-agent: ${{ matrix.config.http-user-agent }}
+          use-public-rspm: true
 
-      - uses: r-lib/actions/setup-pandoc@master
+      - if: runner.os == 'macOS'
+        run: brew install --cask xquartz
```
[the remark this answers is not in the document]
Comment (hadley) [reply .github/workflows/R-CMD-check.yaml:61]: Hmmm, I thought it was, but I'll try removing it to double check.

[the remark this answers is not in the document]
Comment (hadley) [reply .github/workflows/R-CMD-check.yaml]: Confirmed that the macOS build still works without it 😄

[the remark this answers is not in the document]
Comment (yutannihilation) [reply .github/workflows/R-CMD-check.yaml]: Great!

**Comment by yutannihilation** [comment]
Hmm, I haven't seen this error. It seems the change was made recently, but I have no idea what's happening... https://github.com/rstudio/rmarkdown/commit/9d352703818010e6afb672b1a2699ddf5dc8d188

> ```
> could not find function "pandoc_math_engines"
> ```
> (<https://github.com/tidyverse/ggplot2/runs/5409764795?check_suite_focus=true>)

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (hadley) [reply .github/workflows/R-CMD-check.yaml:58]: I think this should happen only rarely, and the advantage of using precompiled binaries is big. (Which is why I switched to ubuntu-latest to get the sf binaries)

Comment (yutannihilation) [reply .github/workflows/R-CMD-check.yaml:58]: I see. Considering the R-devel runner passes here, probably you are right.

**Comment by hadley** [comment]
The `pandoc_math_engines` is a regression in rmarkdown 2.12, is unrelated to this PR, and a fix is in flight. So @yutannihilation if you're ok with it, I think this PR is good to merge.

**Review by yutannihilation — APPROVED** [review]

**Pull request merged**